components/sdmmc/sdmmc_cmd.c

```diff
@@ -606,7 +859,46 @@ static esp_err_t sdmmc_send_cmd_send_csd(sdmmc_card_t* card, sdmmc_csd_t* out_cs
         flip_byte_order(spi_buf,  sizeof(spi_buf));
         ptr = spi_buf;
     }
-    return sdmmc_decode_csd(ptr, out_csd);
+    if (card->host.flags & SDMMC_HOST_MMC_CARD) /* MMC mode */
+        err = mmc_decode_csd(cmd.response, out_csd);
+    else /* SD mode */
+        err = sd_decode_csd(cmd.response, out_csd);
+    return err;
+}
+
+static esp_err_t sdmmc_mem_send_cxd_data(sdmmc_card_t* card , int opcode, void *data, size_t datalen)
+{
+    sdmmc_command_t cmd;
+    void *ptr = NULL;
+    esp_err_t error = ESP_OK;
+
+    ptr = malloc(datalen);
+    if (ptr == NULL) {
+        error = ESP_ERR_NO_MEM;
+        goto out;
+    }
+
+    memset(&cmd, 0, sizeof(cmd));
+    cmd.data = ptr;
+    cmd.datalen = datalen;
+    cmd.blklen = datalen;
+    cmd.opcode = opcode;
+    cmd.arg = 0;
+    cmd.flags = SCF_CMD_ADTC | SCF_CMD_READ;
+    if (opcode == MMC_SEND_EXT_CSD)
```

Comment:
IDF [style guide](http://esp-idf.readthedocs.io/en/latest/contribute/style-guide.html) asks to put braces around one-line bodies of conditionals

(same in a bunch of other places in the code)